_tasks/sync-proposal-data.mjs

```diff
@@ -0,0 +1,324 @@
+  // log.debug({ results });
+  return results[1];
+}
+
+/**
+ * Downloads the proposal's README file using the GitHub API, parses the
+ * Markdown file, and returns the last code block.
+ * @param {string} prpslId
+ * @returns {Array<string>}
+ */
+async function getCodeSample(prpslId) {
+  const codeBlocks = [];
+  try {
+    // Some proposal repos use a different branch for the README.md file (i.e.,
+    // `gh-pages`). Because of this, we need to get some repo metadata before
+    // knowing what the URL is.
+    let rawReadmeUrl;
+    await ghFileImporter
+      .fetchPathMetadata('tc39', prpslId, 'README.md')
+      .then(function (res) {
+        log.info(res.data.download_url);
+        rawReadmeUrl = res.data.download_url;
+      })
+      .catch(function (error) {
+        log.error(error);
+      });
+
+    // Get the file and parse the code samples.
+    await ghFileImporter
+      .importFileFromUrl(rawReadmeUrl)
+      .then(async function (res) {
+        const docSourceText = res;
+        await unified()
+          .use(markdown)
+          .use(function () {
+            return function (tree) {
+              visit(tree, null, (node, index) => {
+                if (node.type === 'code') {
+                  if (
+                    node.lang === 'js' ||
+                    node.lang === 'mjs' ||
+                    node.lang === 'javascript'
+                  ) {
+                    codeBlocks.push(he.encode(node.value));
+                  }
+                }
+              });
+            };
+          })
+          .use(remark2rehype, { allowDangerousHtml: true })
+          .use(raw)
+          .use(htmlStringify)
+          .process(docSourceText);
+      })
+      .catch(function (error) {
+        log.error(error);
+      });
+  } catch (error) {
+    log.error(error);
+  }
+  return codeBlocks;
+}
+
+/**
+ * @param {string} txtVector
+ * @param {number} index
+ * @returns {string} The vector contents.
+ */
+function getTxtVecContent(txtVector, index) {
+  const array = txtVector.split('][');
+  return index === 0 ? array[index].slice(1) : array[index].slice(0, -2);
+}
+
+/**
+ * @param {string} urlKeyLike
+ * @returns {string} The proposal ID.
+ */
+function getPrpslId(urlKeyLike) {
+  let prpslUrl = getUrlFromDoc(urlKeyLike);
+  // Trailing slashes break things and must go.
+  if (prpslUrl.endsWith('/')) {
+    prpslUrl = prpslUrl.slice(0, -1);
+  }
+  // log.debug({ prpslUrl });
+  const prpslId = prpslUrl.split('/').pop().toLowerCase();
+  // log.debug({ prpslId });
+  return prpslId;
+}
+
+/**
+ * Checks if the proposal has a specification by using the files listed by the
+ * GitHub API.
+ * @param {string} prpslId
+ * @returns {boolean}
+ */
+async function hasSpec(prpslId) {
+  // TODO(DerekNonGeneric): This errors out because it tries to fetch a file
+  //     that doesn't exist. It still works, but we need to first check for its
+  //     existence to avoid the error.
+  let isFound = true;
+  try {
```

Comment:
all these try/catches are basically noops since you `.catch` and don't re-throw the error.

Reply:
Yeah, good call. I've updated them to throw until they cause an uncaught exception which triggers an event to display the error message text. I'm not sure how idiomatic this is&hellip;